src/compiler/checker.ts

```diff
@@ -13041,6 +13041,16 @@ namespace ts {
                                     // JsxAttributes has an object-literal flag and undergo same type-assignablity check as normal object-literal.
                                     // However, using an object-literal error message will be very confusing to the users so we give different a message.
                                     // TODO: Spelling suggestions for excess jsx attributes (needs new diagnostic messages)
+
+                                    if (errorNode && isJsxOpeningLikeElement(errorNode.parent)) {
+                                        const attributes = errorNode.parent.attributes;
+                                        for (const jsxProperty of attributes.properties) {

```

Comment:
Instead of searching for this, can you just get the `valueDeclaration` of `prop`?

Reply:
I figured there would be a better way of doing this! The AST node API is just so large it was hard to find where to look for it.